opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -356,6 +361,63 @@ def test_batch_span_processor_not_sampled(self):
         self.assertEqual(len(spans_names_list), 0)
         span_processor.shutdown()
 
+    def _check_fork_trace(self, exporter, expected):
+        time.sleep(0.5)  # give some time for the exporter to upload spans
+        spans = exporter.get_finished_spans()
+        for span in spans:
+            self.assertIn(span.name, expected)
+
+    @unittest.skipUnless(
+        hasattr(os, "fork") and sys.version_info >= (3, 7),
+        "needs *nix and minor version 7 or later",
+    )
+    def test_batch_span_processor_fork(self):
+        tracer_provider = trace.TracerProvider()
+        tracer = tracer_provider.get_tracer(__name__)
+
+        exporter = InMemorySpanExporter()
+        span_processor = export.BatchSpanProcessor(
+            exporter,
+            max_queue_size=256,
+            max_export_batch_size=64,
+            schedule_delay_millis=10,
+        )
+        tracer_provider.add_span_processor(span_processor)
+        with tracer.start_as_current_span("foo"):
+            pass
+        time.sleep(0.5)  # give some time for the exporter to upload spans
+
+        self.assertTrue(span_processor.force_flush())
+        self.assertEqual(len(exporter.get_finished_spans()), 1)
+        exporter.clear()
+        pid = os.fork()
+        if pid:
```

Comment:
I don't think the original process would show this problem regardless of the fix, because the any threads holding the lock will continue to run and eventually release them. The problem is only in the child process IIUC?

You could use [`multiprocessing.Pipe`](https://docs.python.org/3/library/multiprocessing.html#:~:text=and%20process%20safe.-,Pipes,-The%20Pipe()%20function) to send back a success code if the test succeeds in the child process? I guess you'd have to do [`set_start_method('fork')`]( https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method`) beforehand

Reply:
Also should the child process call `os.exit()` so it doesn't continue running the rest of the pytest suite 😅 ?

Reply:
Correct it was only problem in child process. Updated with the suggestion.